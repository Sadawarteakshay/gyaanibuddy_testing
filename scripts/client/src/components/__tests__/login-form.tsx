import React from 'react';
import renderer from 'react-test-renderer';

import { render, cleanup } from '../../test-utils';
import LoginForm from '../login-form';

describe('Login Form', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    const loginFormObject = <LoginForm login={() => {}}/>;

    render(loginFormObject);

    const loginFormTree = renderer
      .create(loginFormObject)
      .toJSON();

    expect(loginFormTree).toMatchSnapshot();
  });

  it('render snapshot', () => {
    const LoginFormElement = <LoginForm login={() => {}}/>;

    render(LoginFormElement);

    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(LoginFormElement)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });
});
