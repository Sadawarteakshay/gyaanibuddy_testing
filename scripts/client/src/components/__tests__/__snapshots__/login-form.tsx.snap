// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form render snapshot 1`] = `
<div
  className="css-12broor"
>
  <header
    className="css-1eziyva"
  >
    <svg
      className="css-185ttn6"
    >
      curve.svg
    </svg>
    <svg
      className="css-4ejw8t"
    >
      logo.svg
    </svg>
  </header>
  <svg
    className="css-1yz0juk"
  >
    rocket.svg
  </svg>
  <h1
    className="css-1r0stgl"
  >
    Space Explorer
  </h1>
  <form
    className="css-ccdq42"
    onSubmit={[Function]}
  >
    <input
      className="css-wotvke"
      data-testid="login-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
    <button
      className="css-wwcn44"
      type="submit"
    >
      Log in
    </button>
  </form>
</div>
`;

exports[`Login Form renders without error 1`] = `
<div
  className="css-12broor"
>
  <header
    className="css-1eziyva"
  >
    <svg
      className="css-185ttn6"
    >
      curve.svg
    </svg>
    <svg
      className="css-4ejw8t"
    >
      logo.svg
    </svg>
  </header>
  <svg
    className="css-1yz0juk"
  >
    rocket.svg
  </svg>
  <h1
    className="css-1r0stgl"
  >
    Space Explorer
  </h1>
  <form
    className="css-ccdq42"
    onSubmit={[Function]}
  >
    <input
      className="css-wotvke"
      data-testid="login-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
    <button
      className="css-wwcn44"
      type="submit"
    >
      Log in
    </button>
  </form>
</div>
`;
